lib/wadler/print.rb

```diff
@@ -115,7 +115,15 @@ def nest(indent, open_obj = '', close_obj = '',
     #
     # @return [Nil]
     def text(value, width: value.length)
-      tokens << Oppen.string(value, width:)
+      if config.trim_trailing_whitespaces? && value.match(/( +)\Z/)
```

Comment:
```suggestion
      if config.trim_trailing_whitespaces? && value.match(/( +)\z/)
```

`\Z` is the same as `$` afaict. `\z` matches the absolute end of string.

I don't know whether we want `\s` here, because `text` could contain `\n`. But I'd say I am more than positive that we want the default white space to also include `\t`. Let's talk about it when you add everything else.

Reply:
> \Z is the same as $ afaict

Not exactly: it accepts end-of-string with or without \n. It's sort of `(?=\n)?\z`